packages/language-server/src/language.ts

```diff
@@ -14,7 +14,7 @@ function virtualFileName(
 ) {
   return `${
     sourceFileName.replace(VINE_FILE_SUFFIX_REGEXP, '')
-  }${extra ? `.${extra}` : ''}.vine-virtual.${extension}`
+  }${`.${extra}` ?? ''}.vine-virtual.${extension}`
```

Comment:
This one is not right actually, because I've made a mistake here before.

the check for `extra` is required, because `.${...}` would never be a falsy value and then fallback to empty string`''`, this would turn to `xxx.undefined.vine-virtual...`

Reply:
这块好像冲突的时候合代码出了问题，我重新拉了最新的代码修改了:p